consensus/ibft/extra_test.go

```diff
@@ -40,3 +58,135 @@ func TestExtraEncoding(t *testing.T) {
 		}
 	}
 }
+
+func generateKeysAndAddresses(t *testing.T, num int) ([]*ecdsa.PrivateKey, []types.Address) {
+	t.Helper()
+
+	keys := make([]*ecdsa.PrivateKey, num)
+	addrs := make([]types.Address, num)
+
+	for i := range keys {
+		pk, addr := tests.GenerateKeyAndAddr(t)
+		keys[i] = pk
+		addrs[i] = addr
+	}
+
+	return keys, addrs
+}
+
+func createIBFTHeader(
+	t *testing.T,
+	num uint64,
+	parentHash types.Hash,
+	validators []types.Address,
+	lastCommittedSeal [][]byte,
+) *types.Header {
+	t.Helper()
+
+	header := &types.Header{
+		Number:     num,
+		ParentHash: parentHash,
+	}
+
+	assert.NoError(t, initIbftExtra(header, validators, lastCommittedSeal))
+
+	return header
+}
+
+// Test Scenario
+// 1. 4 IBFT Validators create headers
+// 2. A faulty node scans the past headers and appends new committed seal
+// 3. Check each hashes of headers are wrong
```

Comment:
Typo here, I'm not sure what you wanted to say

Reply:
This test checks the hash of the block mined by validators is different from the hash of the block including new committed seal by a new validator who joined later.

Reply:
Ah, so the last line (`99`) should be something like:
`Check if each hash of the headers is wrong`

Reply:
Thank you. Fixed at e752d81bb195846d7dd522621a5d8c8d309b7391